Fix Helm chart does not support special characters in access/secret key

dormanze opened the pull request:
## Description
fix #15221 

## Motivation and Context
Fix Helm chart does not support special characters in access/secret key.

## How to test this PR?
prefabricate users with access/secretkey containing special characters.
like:
```
users:
  - accessKey: "tv:(()KW5$`>@/c:dB6_=.4|{jv?@!"
    secretKey: ".B_ES205}Mjzrh}G{emJi'LEuUE#:R"
    policy: consoleAdmin
```
![image](https://user-images.githubusercontent.com/37854724/177665502-5d656d21-10ac-45a0-8c8e-0a384d549dfb.png)

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

dormanze committed:
Update service.yaml

dormanze committed:
Merge branch 'minio:master' into fix_helm

Commit:
fix helm failed

Add two parameters to prevent shell execution failure caused by special characters

dormanze committed:
Merge branch 'minio:master' into fix_helm

Commit:
Merge branch 'fix_helm' of https://github.com/dormanze/minio into fix_helm

dormanze committed:
Update _helper_create_user.txt

upgrade comments.

harshavardhana reviewed (changes requested):
This change is too complicated what is the issue with helm chart not supporting these characters?

Commit:
fix helm failed

Add two parameters to prevent shell execution failure caused by special characters

Commit:
Merge branch 'fix_helm' of https://github.com/dormanze/minio into fix_helm

Commit:
Update _helper_create_user.txt

Add comments

harshavardhana commented on helm/minio/templates/_helper_create_user.txt:
```diff
@@ -35,28 +39,35 @@ connectToMinio() {
+    return 1
+  fi
+  if [[ $(cat $MINIO_ACCESSKEY_SECRETKEY_TMP|wc -l) -ne 2 ]];then
+    echo "tmp file is invalidate."
```
```suggestion
    echo "credentials file is invalid"
```

dormanze replied:
This is a good suggestion.

harshavardhana commented on helm/minio/templates/_helper_create_user.txt:
```diff
@@ -35,28 +39,35 @@ connectToMinio() {
+  POLICY=$1
+  #check accessKey_and_secretKey_tmp file
+  if [[ ! -f $MINIO_ACCESSKEY_SECRETKEY_TMP ]];then
+    echo "not find tmp file."
```
```suggestion
    echo "credentials file does not exist"
```

harshavardhana commented on helm/minio/templates/_helper_create_user.txt:
```diff
@@ -35,28 +39,35 @@ connectToMinio() {
+  if ! checkUserExists ; then
     echo "Creating user '$USER'"
-    ${MC} admin user add myminio $USER $PASS
+    ${MC} admin user add myminio $(head -1 $MINIO_ACCESSKEY_SECRETKEY_TMP) $(tail -1 $MINIO_ACCESSKEY_SECRETKEY_TMP)
```
```suggestion
    cat $MINIO_ACCESSKEY_SECRETKEY_TMP | ${MC} admin user add myminio
```

dormanze replied:
This is a good idea.

harshavardhana commented on helm/minio/templates/_helper_create_user.txt:
```diff
@@ -79,10 +90,13 @@ connectToMinio $scheme
+echo {{ tpl .accessKey $global }} > $MINIO_ACCESSKEY_SECRETKEY_TMP
 {{- if .existingSecret }}
-createUser {{ tpl .accessKey $global }} $(cat /config/secrets/{{ tpl .accessKey $global }}) {{ .policy }}
+cat /config/secrets/{{ tpl .accessKey $global }} >> $MINIO_ACCESSKEY_SECRETKEY_TMP
```
```suggestion
cat /config/secrets/{{ tpl .secretKey $global }} >> $MINIO_ACCESSKEY_SECRETKEY_TMP
```

dormanze replied:
This is a good suggestion.

Commit:
Update _helper_create_user.txt

Modification suggestions.
